scripts/build-zephyr.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+# Imports TARGET, PLATFORM, CPU and TARGET_BUILD_DIR from Makefile
+eval $(make env)
+make info
+
+source $SCRIPT_DIR/settings.sh
+
+# local version of SDK has a higher priority as it might contain a newer SDK
+SDK_LOCAL_LOCATION="$BUILD_DIR/zephyr_sdk"
+if [ -d "$SDK_LOCAL_LOCATION" ]; then
+	export ZEPHYR_SDK_INSTALL_DIR="$SDK_LOCAL_LOCATION"
+elif [ -z "$ZEPHYR_SDK_INSTALL_DIR" ]; then
+	echo "Zephyr SDK not found"
+	echo "Did you forget to run scripts/download-env.sh?"
+	exit 1
+fi
+
+export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
+echo "Using Zephyr SDK from: $ZEPHYR_SDK_INSTALL_DIR"
+
+set -x
+set -e
+
+ZEPHYR_REPO=https://github.com/zephyrproject-rtos/zephyr
+ZEPHYR_REPO_BRANCH=master
+
+case $CPU in
+	vexriscv)
+		if [ "$CPU_VARIANT" != "lite" -a \
+				"$CPU_VARIANT" != "lite+debug" -a \
+				"$CPU_VARIANT" != "standard" -a \
+				"$CPU_VARIANT" != "standard+debug" -a \
+				"$CPU_VARIANT" != "full" -a \
+				"$CPU_VARIANT" != "full+debug" -a \
+				"$CPU_VARIANT" != "linux" ]; then
+			echo "Zephyr needs a CPU_VARIANT set to at least 'lite' for the support of 'ecall' instruction."
+			echo "Supported variants: lite, lite+debug, standard, standard+debug, full, full+debug, linux."
+			exit 1
+		fi
+		TARGET_BOARD=arty_litex_vexriscv
+		ZEPHYR_REPO=https://github.com/antmicro/zephyr
+		ZEPHYR_REPO_BRANCH=litex-vexriscv
+		;;
+	*)
+		echo "CPU $CPU isn't supported at the moment."
+		exit 1
+		;;
+esac
+
+if [ "$FIRMWARE" != "zephyr" ]; then
+	echo "When building Zephyr you should set FIRMWARE to 'zephyr'."
+	exit 1
+fi
+
+function check_python_package {
+	python -c "import $1"
+}
+
+check_python_package "yaml"     || conda install pyyaml
```

Comment:
These three python packages should maybe be part of the `download-env.sh` script too?

Reply:
moved